### PR TITLE
[DOCS] adding the link checker to the build

### PR DESCRIPTION
**Pull request by donaldheppner**
Changes proposed in this pull request:
- Adding checking the internal links as part of the CI for docs; if a broken link is added to docs, the build will fail

### CODE REVIEW

File: azure-pipelines-docs-integration.yml [hunk 1]
```diff
@@ -55,6 +55,15 @@ stages:
           - bash: ./scripts/check_for_docs_deps_changes
             name: CheckDocsDependenciesChanges
 
+  - stage: custom_checks
+    pool:
+      vmImage: 'ubuntu-latest'
+    jobs:
+    - job: link_checker
+      condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckDocsChanges.DocsChanged'], true), eq(variables.isMain, true))
```
Comment (cdkini) [inline azure-pipelines-docs-integration.yml:63]: Should we also add `CheckDocsDependenciesChanges` to this condition? See L72 below for reference.

This is a script that determines whether or not source code in GE has changed that might have an impact on our docs under test.

Comment (donaldheppner) [reply azure-pipelines-docs-integration.yml:63]: I left it out because I didn't think those dependencies could result in broken links

Comment (cdkini) [reply azure-pipelines-docs-integration.yml:63]: You are 100% correct disregard that comment!

File: azure-pipelines-docs-integration.yml [hunk 2]
```diff
@@ -55,6 +55,15 @@ stages:
...
+      steps:
+      - bash: python scripts/docs_link_checker.py -p docs -r docs -s docs --skip-external
+
```
Comment (cdkini) [inline azure-pipelines-docs-integration.yml:66]: might we want to add a `name` here so it doesn't show up as `Bash` in CI/CD? I'm good either way.

**Review by cdkini — APPROVED** [review]
Beautiful! Thanks for this 🙇🏽

**Pull request merged**